dashboard/src/components/triggers/tableDataWrapper.test.js

```diff
@@ -0,0 +1,347 @@
+import { ThemeProvider } from '@mui/material/styles';
+import { cleanup, fireEvent, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import 'jest-canvas-mock';
+import React from 'react';
+import TEST_THEME from '../../styles/theme';
+import TableDataWrapper from './tableDataWrapper';
+import { triggerDetail, triggerListResponse } from './test_data';
+import {
+  CONFIRM_DELETE,
+  DELETE_FINAL_WARNING,
+  DELETE_WARNING,
+  PRE_DELETE,
+} from './triggerDetail';
+
+const defaultMock = Object.freeze({
+  getTriggers: jest.fn().mockResolvedValue(triggerListResponse),
+  postTrigger: jest.fn(),
+  getTriggerDetails: jest.fn().mockResolvedValue(triggerDetail),
+  deleteTrigger: jest.fn().mockResolvedValue(true),
+});
+let dataAPIMock = {};
+
+jest.mock('../../hooks/dataAPI', () => ({
+  useDataAPI: () => {
+    return dataAPIMock;
+  },
+}));
+
+describe('Trigger table data wrapper tests', () => {
+  const DIV_NODE = 'DIV';
+  const P_NODE = 'P';
+  const SEARCH_INPUT_ID = 'searcInputId';
+  const NEW_TRIGGER_BTN_ID = 'newTriggerId';
+  const NEW_TRIGGER_SAVE_ID = 'saveNewTriggerId';
+  const CREATE_TITLE = 'Create New Trigger';
+  const USER_EVENT_ENTER = '{enter}';
+  const USER_EVENT_DELETE = '{backspace}';
+  const VALIDATION_ERROR = 'Enter a value.';
+  const TRIGGER_NAME_INPUT_ID = 'triggerNameInputId';
+  const TRIGGER_SCHEDULE_INPUT_ID = 'triggerScheduleInputId';
+  const DETAIL_TYPE_ID = 'detailTypeId';
+  const DETAIL_SCHEDULE_ID = 'detailScheduleId';
+  const DETAIL_OWNER_ID = 'detailOwnerId';
+  const DELETE_TRIGGER_BTN_ID = 'deleteTriggerBtnId';
+  const ROW_DELETE_BTN_ = 'triggerDelete-';
+  const DELETE_WARNING_ID = 'deleteWarning';
+  const DELETE_FINAL_ID = 'deleteFinal';
+
+  // this is just a little hack to silence a warning that we'll get until we
+  // upgrade to 16.9. See also: https://github.com/facebook/react/pull/14853
```

Comment:
any reason we don't upgrade?

Reply:
yeah waiting on the dashboard redesign to overhaul a lot of this. 

don't want to kick the hornet's nest by moving up react and all the other libs in one go